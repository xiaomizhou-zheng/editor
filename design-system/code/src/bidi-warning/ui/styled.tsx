/** @jsx jsx */
import { ReactNode } from 'react';

import { css, jsx } from '@emotion/react';

import { Y75 } from '@atlaskit/theme/colors';
import { token } from '@atlaskit/tokens';

const decoration = css({
  // Required as otherwise the following bidi characters cause the span
  // to not receive hover events.
  //
  // U+2066 LEFT-TO-RIGHT ISOLATE (when using pseudo element before)
  // U+202E RIGHT-TO-LEFT OVERRIDE' (when using pseudo element after)
  position: 'relative',

  ':before': {
    display: 'inline-flex',
    padding: `${token('spacing.scale.0', '0px')} ${token(
      'spacing.scale.050',
      '4px',
    )}`,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    background: token('color.background.warning', Y75),
    color: token('color.text.warning', '#7F5F01'),
    content: '"<"attr(data-bidi-character-code)">"',
    fontSize: '14px',
    fontStyle: 'normal',
    lineHeight: '18px',
    /**
     * Ensures the decoration receives pointer events when it occurs with
     * an ancestor that disables them.
     */
    pointerEvents: 'auto',
  },

  ':hover:before': {
    background: token('color.background.warning.hovered', Y75),
    color: token('color.text.warning', '#533F04'),
  },
});

export function Decorator({
  bidiCharacter,
  children,
  testId,
}: {
  bidiCharacter: string;
  children: ReactNode;
  testId?: string;
}) {
  const bidiCharacterCode = getBidiCharacterCode(bidiCharacter);
  return (
    <span aria-label={bidiCharacterCode}>
      <span
        css={decoration}
        data-testid={testId}
        data-bidi-character-code={bidiCharacterCode}
        // This is set to true so that the content is not read out by
        // screen readers as the content includes angle brackets for
        // visual decoration purposes.
        // We use a span with the aria-label set to the bidi character  code
        // above this span for screen readers.
        aria-hidden="true"
      >
        {children}
      </span>
    </span>
  );
}

function getBidiCharacterCode(bidiCharacter: string) {
  const bidiCharacterCode = bidiCharacter.codePointAt(0)?.toString(16);

  return `U+${bidiCharacterCode}`;
}
