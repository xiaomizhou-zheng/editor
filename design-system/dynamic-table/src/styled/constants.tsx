import { LegacyRef, ReactNode } from 'react';

import { css } from '@emotion/react';

import { token } from '@atlaskit/tokens';

const CSS_VAR_WIDTH = '--local-dynamic-table-width';

export interface TruncateStyleProps {
  width?: number;
  isFixedSize?: boolean;
  shouldTruncate?: boolean;
  children?: ReactNode;
  testId?: string;
  innerRef?: LegacyRef<HTMLTableCellElement | HTMLTableRowElement> | undefined;
  className?: string;
}

export const truncationWidthStyles = css({ width: `var(${CSS_VAR_WIDTH})` });

export const fixedSizeTruncateStyles = css({
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap',
});

export const overflowTruncateStyles = css({
  overflow: 'hidden',
});

export const getTruncationStyleVars = ({ width }: TruncateStyleProps) =>
  typeof width !== 'undefined' ? { [CSS_VAR_WIDTH]: `${width}%` } : undefined;

export const cellStyles = css({
  padding: `${token('spacing.scale.050', '4px')} ${token(
    'spacing.scale.100',
    '8px',
  )}`,
  border: 'none',
  textAlign: 'left',
  '&:first-of-type': {
    // TODO Delete this comment after verifying spacing token -> previous value `0`
    paddingLeft: token('spacing.scale.0', '0px'),
  },
  '&:last-child': {
    // TODO Delete this comment after verifying spacing token -> previous value `0`
    paddingRight: token('spacing.scale.0', '0px'),
  },
});
