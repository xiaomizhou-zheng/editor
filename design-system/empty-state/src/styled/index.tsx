export { default as ActionsContainer } from './actions-container';
export { default as Container } from './container';
export { default as Description } from './description';
export { default as Header } from './header';
export { default as Image } from './image';
export { default as SpinnerContainer } from './spinner-container';
