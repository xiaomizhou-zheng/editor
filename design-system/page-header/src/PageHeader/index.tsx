// TODO: Remove this eslint-disable when prop names have been renamed.
// This rule is here as prop name changes are a major as they are used
// by our consumers. This can be done in the next lite-mode conversion.
/* eslint-disable @repo/internal/react/boolean-prop-naming-convention */
import React, { ReactElement, ReactNode } from 'react';

import {
  ActionsWrapper,
  BottomBarWrapper,
  OuterWrapper,
  StyledTitleWrapper,
  TitleContainer,
  TitleWrapper,
} from './styled';

export type PageHeaderProps = {
  /**
   * Contents of the action bar to be rendered next to the page title.
   */
  actions?: ReactElement;
  /**
   * Contents of the action bar to be rendered next to the page title. Typically a button group.
   */
  bottomBar?: ReactElement;
  /**
   * Page breadcrumbs to be rendered above the title.
   */
  breadcrumbs?: ReactElement;
  /**
   * Contents of the bottom bar to be rendered below the page title. Typically contains a search bar and/or filters.
   */
  children?: ReactNode;
  /**
   * Content of the page title. The text wraps by default.
   */
  disableTitleStyles?: boolean;
  /**
   * Returns the inner ref to the DOM element of the title. This is exposed so the focus can be set.
   */
  innerRef?: (element: HTMLElement) => void;
  /**
   * Prevent the title from wrapping across lines. This should be avoided.
   */
  truncateTitle?: boolean;
  /**
   * Used as id of inner h1 tag. This is exposed so the header text can be used as label of other element by aria-labeledby
   */
  id?: string;
};

/**
 * __Page header__
 *
 * A page header defines the top of a page. It contains a title and can be optionally combined with
 * breadcrumbs buttons, search, and filters.
 *
 * - [Examples](https://atlassian.design/components/page-header/examples)
 * - [Code](https://atlassian.design/components/page-header/code)
 * - [Usage](https://atlassian.design/components/page-header/usage)
 */
const PageHeader = ({
  innerRef,
  breadcrumbs,
  actions,
  bottomBar,
  children,
  id,
  disableTitleStyles = false,
  truncateTitle = false,
}: PageHeaderProps) => {
  return (
    <OuterWrapper>
      {breadcrumbs}
      <TitleWrapper truncateTitle={truncateTitle}>
        <TitleContainer truncateTitle={truncateTitle}>
          {disableTitleStyles ? (
            children
          ) : (
            <StyledTitleWrapper
              ref={innerRef as React.Ref<HTMLHeadingElement>}
              truncateTitle={truncateTitle}
              id={id}
            >
              {children}
            </StyledTitleWrapper>
          )}
        </TitleContainer>
        {actions && <ActionsWrapper>{actions}</ActionsWrapper>}
      </TitleWrapper>
      {bottomBar && <BottomBarWrapper> {bottomBar} </BottomBarWrapper>}
    </OuterWrapper>
  );
};

export default PageHeader;
