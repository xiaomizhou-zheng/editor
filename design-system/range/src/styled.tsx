/** @jsx jsx */

import { CSSProperties, forwardRef } from 'react';

import { css, jsx } from '@emotion/react';

import { token } from '@atlaskit/tokens';

import * as theme from './theme';

type InputProps = React.InputHTMLAttributes<HTMLInputElement> & {
  ref: React.Ref<HTMLInputElement>;
  valuePercent: string;
};

const VAR_THUMB_BORDER_COLOR = '--thumb-border';
const VAR_THUMB_SHADOW = '--thumb-shadow';
const VAR_THUMB_BACKGROUND_COLOR = '--thumb-bg';
const VAR_TRACK_BACKGROUND_COLOR = '--track-bg';
const VAR_TRACK_FOREGROUND_COLOR = '--track-fg';
const VAR_TRACK_FOREGROUND_WIDTH = '--track-fg-width';

const sliderThumbStyles = {
  boxSizing: 'border-box',
  width: theme.thumb.size,
  height: theme.thumb.size,
  border: 'none',
  background: `var(${VAR_THUMB_BACKGROUND_COLOR}, ${theme.thumb.background.default})`,
  // adapted from @atlaskit/focus-ring
  outline: `solid 2px var(${VAR_THUMB_BORDER_COLOR})`,
  outlineOffset: '2px',
  borderRadius: '50%',
  boxShadow: `var(${VAR_THUMB_SHADOW})`,
  cursor: 'pointer', // Different implicit behavior across browsers -> making it explicit.
  transition: `background-color ${theme.transitionDuration} ease-in-out`,
} as const;

const sliderTrackStyles = {
  borderRadius: theme.track.borderRadius,
  border: 0,
  cursor: 'pointer',
  height: theme.track.height,
  width: '100%',
  transition: `background-color ${theme.transitionDuration} ease-in-out`,
} as const;

// Styles are split per browser as they are implemented differently
// Cannot consolidate as Chrome & Firefox don't recognise styles if they are grouped
// with CSS selectors they don't recognise, e.g. &::-ms-thumb
const browserStyles = {
  webkit: css({
    WebkitAppearance: 'none', // Hides the slider so that custom slider can be made
    '::-webkit-slider-thumb': {
      ...sliderThumbStyles,
      marginTop: -(theme.thumb.size - theme.track.height) / 2,
      WebkitAppearance: 'none',
    },
    '::-webkit-slider-runnable-track': {
      ...sliderTrackStyles,
      /**
       * Webkit does not have separate properties for the background/foreground like firefox.
       * Instead we use background layering:
       * - The gray background is a simple background color.
       * - The blue foreground is a 'gradient' (to create a color block) that is sized to the progress.
       *
       * Individual properties have been used over the `background` shorthand for readability.
       */
      backgroundColor: `var(${VAR_TRACK_BACKGROUND_COLOR})`,
      backgroundImage: `linear-gradient(var(${VAR_TRACK_FOREGROUND_COLOR}), var(${VAR_TRACK_FOREGROUND_COLOR}))`,
      backgroundRepeat: 'no-repeat',
      backgroundSize: `var(${VAR_TRACK_FOREGROUND_WIDTH}) 100%`,
      // eslint-disable-next-line @repo/internal/styles/no-nested-styles
      '[dir="rtl"] &': {
        backgroundPosition: 'right',
      },
    },
    ':disabled': {
      '::-webkit-slider-thumb, ::-webkit-slider-runnable-track': {
        cursor: 'not-allowed',
      },
    },
  }),
  firefox: css({
    '::-moz-range-thumb': sliderThumbStyles,
    '::-moz-focus-outer': {
      border: 0,
    },
    '::-moz-range-progress': {
      ...sliderTrackStyles,
      background: `var(${VAR_TRACK_FOREGROUND_COLOR})`,
    },
    '::-moz-range-track': {
      ...sliderTrackStyles,
      background: `var(${VAR_TRACK_BACKGROUND_COLOR})`,
    },
    ':disabled': {
      '::-moz-range-thumb, ::-moz-range-progress, ::-moz-range-track': {
        cursor: 'not-allowed',
      },
    },
  }),
};

const baseStyles = css({
  width: '100%', // Has a fixed width by default
  height: theme.input.height, // Otherwise thumb will collide with previous box element
  background: 'transparent', // Otherwise white
  ':focus': {
    outline: 'none',
  },
  ':disabled': {
    cursor: 'not-allowed',
    opacity: token('opacity.disabled', '0.4'),
  },
});

const themeStyles = css({
  [VAR_THUMB_SHADOW]: theme.thumb.boxShadow.default,
  [VAR_TRACK_BACKGROUND_COLOR]: theme.track.background.default,
  [VAR_TRACK_FOREGROUND_COLOR]: theme.track.foreground.default,
  ':hover:not(:disabled)': {
    [VAR_THUMB_BACKGROUND_COLOR]: theme.thumb.background.hovered,
    [VAR_TRACK_BACKGROUND_COLOR]: theme.track.background.hovered,
    [VAR_TRACK_FOREGROUND_COLOR]: theme.track.foreground.hovered,
  },
  ':active:not(:disabled)': {
    [VAR_THUMB_BACKGROUND_COLOR]: theme.thumb.background.pressed,
  },
  ':focus-visible': {
    [VAR_THUMB_BORDER_COLOR]: theme.thumb.borderColor.focused,
  },
});

/**
 * __Input__
 * Internal-only styled input component.
 */
export const Input = forwardRef(
  (props: InputProps, ref: React.Ref<HTMLInputElement>) => {
    const { valuePercent, style, ...strippedProps } = props;

    return (
      <input
        {...strippedProps}
        style={
          {
            // We are creating a css variable to control the "progress" portion of the range input
            // This avoids us needing to create a new css class for each new percentage value
            [VAR_TRACK_FOREGROUND_WIDTH]: `${valuePercent}%`,
          } as CSSProperties
        }
        ref={ref}
        css={[
          baseStyles,
          browserStyles.webkit,
          browserStyles.firefox,
          themeStyles,
        ]}
      />
    );
  },
);

Input.displayName = 'InputRange';
