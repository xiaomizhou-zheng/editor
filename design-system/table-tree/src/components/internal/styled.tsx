/** @jsx jsx */
import { FC, HTMLAttributes } from 'react';

import { css, jsx } from '@emotion/react';

import { N30, N800 } from '@atlaskit/theme/colors';
import { token } from '@atlaskit/tokens';

export const iconColor = token('color.text', N800);

const treeRowContainerStyles = css({
  display: 'flex',
  borderBottom: `1px solid ${token('color.border', N30)}`,
});

/**
 * __Tree row container__
 */
export const TreeRowContainer: FC<HTMLAttributes<HTMLDivElement>> = (props) => (
  // eslint-disable-next-line @repo/internal/react/no-unsafe-spread-props
  <div css={treeRowContainerStyles} {...props} />
);

const commonChevronContainerStyles = css({
  display: 'flex',
  marginLeft: -25,
  position: 'absolute',
  top: 7,
  alignItems: 'center',
});

/**
 * __Chevron container__
 */
export const ChevronContainer: FC<HTMLAttributes<HTMLSpanElement>> = (
  props,
  // eslint-disable-next-line @repo/internal/react/no-unsafe-spread-props
) => <span {...props} css={commonChevronContainerStyles} />;

const chevronIconContainerStyles = css({
  position: 'relative',
  top: 1,
});

/**
 * __Chevron icon container__
 *
 * A chevron icon container.
 */
export const ChevronIconContainer: FC<HTMLAttributes<HTMLSpanElement>> = (
  props,
  // eslint-disable-next-line @repo/internal/react/no-unsafe-spread-props
) => <span {...props} css={chevronIconContainerStyles} />;

const loadingItemContainerStyles = css({
  width: '100%',
  paddingTop: 5,
});

const paddingLeftStyles = css({
  paddingLeft: '50%',
});

/**
 * __Loader item container__
 *
 * A loader item container.
 */
export const LoaderItemContainer: FC<{ isRoot?: boolean }> = ({
  isRoot,
  ...props
}) => (
  <span
    css={[
      commonChevronContainerStyles,
      loadingItemContainerStyles,
      isRoot && paddingLeftStyles,
    ]}
    // eslint-disable-next-line @repo/internal/react/no-unsafe-spread-props
    {...props}
  />
);
