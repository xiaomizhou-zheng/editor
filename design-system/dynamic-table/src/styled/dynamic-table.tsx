/* eslint-disable @repo/internal/react/require-jsdoc */
/** @jsx jsx */
import { FC, forwardRef, HTMLProps } from 'react';

import { css, jsx } from '@emotion/react';

import { gridSize as getGridSize } from '@atlaskit/theme/constants';
import { token } from '@atlaskit/tokens';

import { row, tableBorder } from '../theme';

const gridSize = getGridSize();

export type TableProps = HTMLProps<HTMLTableElement> & {
  isFixedSize?: boolean;
  hasDataRow: boolean;
};

// CSS vars for table row
// these are declared here to avoid being re-declared in each table row
export const tableRowCSSVars = {
  CSS_VAR_HOVER_BACKGROUND: '--local-dynamic-table-hover-bg',
  CSS_VAR_HIGHLIGHTED_BACKGROUND: '--local-dynamic-table-highlighted-bg',
  CSS_VAR_HOVER_HIGHLIGHTED_BACKGROUND:
    '--local-dynamic-table-hover-highlighted-bg',
  CSS_VAR_ROW_FOCUS_OUTLINE: '--local-dynamic-table-row-focus-outline',
};

const fixedSizeTableStyles = css({
  tableLayout: 'fixed',
});

const tableStyles = css({
  width: '100%',
  borderCollapse: 'separate',
  borderSpacing: '0px',
});

const bodyBorder = css({
  borderBottom: `2px solid ${tableBorder.borderColor}`,
});

export const Table = forwardRef<HTMLTableElement, TableProps>(
  ({ isFixedSize, hasDataRow, children, ...rest }, ref) => {
    return (
      <table
        style={
          {
            [tableRowCSSVars.CSS_VAR_HOVER_BACKGROUND]: row.hoverBackground,
            [tableRowCSSVars.CSS_VAR_HIGHLIGHTED_BACKGROUND]:
              row.highlightedBackground,
            [tableRowCSSVars.CSS_VAR_HOVER_HIGHLIGHTED_BACKGROUND]:
              row.hoverHighlightedBackground,
            [tableRowCSSVars.CSS_VAR_ROW_FOCUS_OUTLINE]: row.focusOutline,
          } as React.CSSProperties
        }
        css={[
          tableStyles,
          isFixedSize && fixedSizeTableStyles,
          hasDataRow && bodyBorder,
        ]}
        ref={ref}
        {...rest}
      >
        {children}
      </table>
    );
  },
);

const captionStyles = css({
  marginTop: `${gridSize * 3.5}px`,
  // TODO Delete this comment after verifying spacing token -> previous value ``${gridSize}px``
  marginBottom: token('spacing.scale.100', '8px'),
  fontSize: '1.42857143em',
  fontStyle: 'inherit',
  fontWeight: 500,
  letterSpacing: '-0.008em',
  lineHeight: 1.2,
  textAlign: 'left',
  /* there is a bug in Safari: if element which creates a new stacking context
    is a child of a table, table caption re-renders in bad wrong position
    https://stackoverflow.com/questions/44009186/safari-bug-translating-table-row-group-using-gsap-make-caption-jump-to-bottom
  */
  willChange: 'transform',
});

export const Caption: FC = ({ children }) => (
  <caption css={captionStyles}>{children}</caption>
);

const paginationWrapperStyles = css({
  display: 'flex',
  justifyContent: 'center',
});

export const PaginationWrapper: FC = ({ children }) => (
  <div css={paginationWrapperStyles}>{children}</div>
);
