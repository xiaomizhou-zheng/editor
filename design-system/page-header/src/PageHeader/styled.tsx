// TODO: Remove this eslint-disable when prop names have been renamed.
// This rule is here as prop name changes are a major as they are used
// by our consumers. The prop name concerned here is truncateTitle.
// This can be done in the next lite-mode conversion.
/* eslint-disable @repo/internal/react/boolean-prop-naming-convention */
/** @jsx jsx */

import React from 'react';

import { css, jsx } from '@emotion/react';

import { gridSize as getGridSize } from '@atlaskit/theme/constants';
import { h700 } from '@atlaskit/theme/typography';

const gridSize = getGridSize();

const truncateStyles = css({
  overflowX: 'hidden',
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap',
});

const outerStyles = css({
  margin: `${gridSize * 3}px 0 ${gridSize * 2}px 0`,
});

const styledTitleStyles = css({
  marginTop: 0,
  lineHeight: `${gridSize * 4}px`,
  outline: 'none',
});

const titleWrapperStyles = css({
  display: 'flex',
  alignItems: 'flex-start',
  flexWrap: 'wrap',
});

const titleWrapperTruncateStyles = css({
  flexWrap: 'nowrap',
});

const titleContainerStyles = css({
  minWidth: 0,
  maxWidth: '100%',
  marginBottom: `${gridSize}px`,
  flex: '1 0 auto',
  flexShrink: undefined,
});

const actionStyles = css({
  maxWidth: '100%',
  marginBottom: `${gridSize}px`,
  marginLeft: 'auto',
  paddingLeft: `${gridSize * 4}px`,
  flex: '0 0 auto',
  whiteSpace: 'nowrap',
  // eslint-disable-next-line @repo/internal/styles/no-nested-styles
  '>': {
    textAlign: 'right',
  },
});

const titleContainerTruncateStyles = css({
  flexShrink: 1,
});

const bottomBarStyles = css({
  marginTop: `${gridSize * 2}px`,
});

/**
 * __Outer wrapper__
 *
 * An outer wrapper that is the outermost component of the PageHeader component. It wraps around the PageHeader, its Actions,
 * the BottomBar and its Breadcrumbs.
 *
 */
export const OuterWrapper: React.FC<React.ReactNode> = ({ children }) => {
  return <div css={outerStyles}>{children}</div>;
};

// eslint-disable-next-line @repo/internal/react/consistent-css-prop-usage
const h700Styles = css(h700());

interface StyledTitleWrapperProps {
  children?: React.ReactNode;
  id?: string;
  truncateTitle?: boolean;
}

/**
 * __Styled title wrapper__
 *
 * A styled title wrapper is a wrapper around the title that controls its the styles exclusively.
 *
 */
export const StyledTitleWrapper = React.forwardRef<
  HTMLHeadingElement,
  StyledTitleWrapperProps
>(({ children, id, truncateTitle }, ref) => {
  return (
    <h1
      css={[h700Styles, styledTitleStyles, truncateTitle && truncateStyles]}
      ref={ref}
      tabIndex={-1}
      id={id}
    >
      {children}
    </h1>
  );
});

interface TitleProps {
  truncateTitle?: boolean;
}

/**
 * __Title wrapper__
 *
 * A title wrapper is a wrapper around the title and the actions.
 *
 */
export const TitleWrapper: React.FC<TitleProps> = ({
  children,
  truncateTitle,
}) => {
  return (
    <div
      css={[titleWrapperStyles, truncateTitle && titleWrapperTruncateStyles]}
    >
      {children}
    </div>
  );
};

/**
 * Title container
 *
 * A title container is a container that wraps around the title and its styles (if applied).
 *
 */
export const TitleContainer: React.FC<TitleProps> = ({
  children,
  truncateTitle,
}) => {
  return (
    <div
      css={[
        titleContainerStyles,
        truncateTitle && titleContainerTruncateStyles,
      ]}
    >
      {children}
    </div>
  );
};

/**
 * __Actions wrapper__
 *
 * An actions wrapper is a wrapper for the actions, which appear on the top right of the PageHeader component.
 *
 */
export const ActionsWrapper: React.FC = ({ children }) => {
  return <div css={actionStyles}>{children}</div>;
};

/**
 * __Bottom bar wrapper__
 *
 * A bottom bar wrapper is a wrapper for the bottom bar, which appears at the bottom of the PageHeader component.
 *
 */
export const BottomBarWrapper: React.FC = ({ children }) => {
  return <div css={bottomBarStyles}>{children}</div>;
};
