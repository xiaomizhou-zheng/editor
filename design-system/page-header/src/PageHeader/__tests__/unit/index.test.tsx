import React from 'react';

import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import PageHeader from '../../../index';
import { StyledTitleWrapper } from '../../styled';

describe('@atlaskit/page-header', () => {
  it('should render correctly', () => {
    const BreadCrumbs = () => <div>Breadcrumb</div>;
    const Actions = () => <div>Action</div>;
    const Bar = () => <div>Bar</div>;

    const wrapper = (
      <PageHeader
        breadcrumbs={<BreadCrumbs />}
        actions={<Actions />}
        bottomBar={<Bar />}
      >
        Test
      </PageHeader>
    );
    const Component = renderer.create(wrapper).toJSON();
    expect(Component).toMatchSnapshot();
  });

  it('should render correctly with disableTitleStyles prop', () => {
    const BreadCrumbs = () => <div>Breadcrumb</div>;
    const Actions = () => <div>Action</div>;
    const Bar = () => <div>Bar</div>;

    const wrapper = (
      <PageHeader
        breadcrumbs={<BreadCrumbs />}
        actions={<Actions />}
        bottomBar={<Bar />}
        disableTitleStyles
      >
        Test
      </PageHeader>
    );

    const Component = renderer.create(wrapper).toJSON();
    expect(Component).toMatchSnapshot();
  });

  it('should render passed children', () => {
    const wrapper = shallow(<PageHeader>Title</PageHeader>);
    expect(wrapper.contains('Title')).toBe(true);
  });

  it('should render passed breadcrumbs', () => {
    const BreadCrumbs = () => <div>Breadcrumb</div>;
    const wrapper = shallow(
      <PageHeader breadcrumbs={<BreadCrumbs />}>Title</PageHeader>,
    );
    expect(wrapper.find(BreadCrumbs).length).toBe(1);
  });

  it('should render passed actions', () => {
    const Actions = () => <div>Breadcrumb</div>;
    const wrapper = shallow(
      <PageHeader actions={<Actions />}>Title</PageHeader>,
    );
    expect(wrapper.find(Actions).length).toBe(1);
  });

  it('should render passed bottom bar', () => {
    const Bar = () => <div>Breadcrumb</div>;
    const wrapper = shallow(<PageHeader bottomBar={<Bar />}>Title</PageHeader>);
    expect(wrapper.find(Bar).length).toBe(1);
  });

  it('should render custom component without the StyledTitle when disableTitleStyles is true', () => {
    const CustomTitle = () => <span>Custom component</span>;
    const wrapper = shallow(
      <PageHeader disableTitleStyles>
        <CustomTitle />
      </PageHeader>,
    );
    expect(wrapper.find(StyledTitleWrapper)).toHaveLength(0);
  });

  it('should truncate with truncateTitle prop', () => {
    const wrapper = mount(
      <PageHeader truncateTitle>Long heading text</PageHeader>,
    );

    const element = wrapper.find('h1');

    expect(element).toHaveStyleDeclaration('white-space', 'nowrap');
    expect(element).toHaveStyleDeclaration('text-overflow', 'ellipsis');
    expect(element).toHaveStyleDeclaration('overflow-x', 'hidden');
  });

  it('should set received id prop as id of inner h1 element', () => {
    const wrapper = mount(
      <PageHeader id="page-heading">Long heading text</PageHeader>,
    );

    expect(wrapper.find('h1').getDOMNode()).toHaveAttribute(
      'id',
      'page-heading',
    );
  });
});
