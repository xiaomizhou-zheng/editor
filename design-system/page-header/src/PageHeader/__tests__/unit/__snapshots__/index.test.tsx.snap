// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@atlaskit/page-header should render correctly 1`] = `
.emotion-5 {
  margin: 24px 0 16px 0;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-1 {
  min-width: 0;
  max-width: 100%;
  margin-bottom: 8px;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.emotion-0 {
  font-size: 1.7142857142857142em;
  font-style: inherit;
  line-height: 1.1666666666666667;
  color: var(--ds-text, var(--ds-text, #172B4D));
  font-weight: 500;
  letter-spacing: -0.01em;
  margin-top: 40px;
  margin-top: 0;
  line-height: 32px;
  outline: none;
}

.emotion-2 {
  max-width: 100%;
  margin-bottom: 8px;
  margin-left: auto;
  padding-left: 32px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  white-space: nowrap;
}

.emotion-2 > {
  text-align: right;
}

.emotion-4 {
  margin-top: 16px;
}

<div
  className="emotion-5"
>
  <div>
    Breadcrumb
  </div>
  <div
    className="emotion-3"
  >
    <div
      className="emotion-1"
    >
      <h1
        className="emotion-0"
        tabIndex={-1}
      >
        Test
      </h1>
    </div>
    <div
      className="emotion-2"
    >
      <div>
        Action
      </div>
    </div>
  </div>
  <div
    className="emotion-4"
  >
     
    <div>
      Bar
    </div>
     
  </div>
</div>
`;

exports[`@atlaskit/page-header should render correctly with disableTitleStyles prop 1`] = `
.emotion-4 {
  margin: 24px 0 16px 0;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-0 {
  min-width: 0;
  max-width: 100%;
  margin-bottom: 8px;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.emotion-1 {
  max-width: 100%;
  margin-bottom: 8px;
  margin-left: auto;
  padding-left: 32px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  white-space: nowrap;
}

.emotion-1 > {
  text-align: right;
}

.emotion-3 {
  margin-top: 16px;
}

<div
  className="emotion-4"
>
  <div>
    Breadcrumb
  </div>
  <div
    className="emotion-2"
  >
    <div
      className="emotion-0"
    >
      Test
    </div>
    <div
      className="emotion-1"
    >
      <div>
        Action
      </div>
    </div>
  </div>
  <div
    className="emotion-3"
  >
     
    <div>
      Bar
    </div>
     
  </div>
</div>
`;
